packages/format-currency/src/index.js

```diff
@@ -58,11 +58,11 @@ export function getCurrencyObject( number, code, options = {} ) {
 	const sign = number < 0 ? '-' : '';
 	const absNumber = Math.abs( number );
 	const rawInteger = Math.floor( absNumber );
-	const integer = numberFormat( rawInteger, {
-		decimals: 0,
+	const integer = numberFormat( absNumber, {
+		decimals: precision,
 		thousandsSep: grouping,
 		decPoint: decimal,
-	} );
+	} ).split( decimal )[ 0 ];
 	const fraction =
 		precision > 0
 			? numberFormat( absNumber - rawInteger, {
```

Comment:
44e56adbfd07c622bd18d8ddee8d88b3ca672197 adds support for 0.005 rounding in `numberFormat()`, but doing loose math on unknown numbers here can still cause the same rounding issue to occur. 

For example, if `absNumber` is `1.005`, the result of `absNumber - rawInteger` is `0.004999999...` which will get rounded down to `0.00` when `precision` is `2`. I think that the same rounding solution would need to be applied to any math that has potential rounding prior to passing the result into `numberFormat` 🤔